doc/source/parameters/cfd/linear_solver_control.rst

```diff
@@ -0,0 +1,157 @@
+
+	  #---------------------------------------------------
+	  # ILU smoother/coarsener parameters for AMG preconditioner
+	  #---------------------------------------------------
+	  # AMG preconditioner ILU smoother/coarsener fill
+	  set amg preconditioner ilu fill = 0
+
+	  # AMG preconditioner ILU smoother/coarsener absolute tolerance
+	  set amg preconditioner ilu absolute tolerance = 1e-12
+
+	  # AMG preconditioner ILU smoother/coarsener relative tolerance
+	  set amg preconditioner ilu relative tolerance = 1.00
+
+	  #---------------------------------------------------
+	  # other AMG solver parameters
+	  #---------------------------------------------------
+	  # AMG aggregation threshold
+	  set amg aggregation threshold = 1e-14
+
+	  # AMG number of cycles
+	  set amg n cycles              = 1
+
+	  # AMG w cycling. If this is set to true, W cycling is used. Otherwise, V cycling is used.
+	  set amg w cycles              = false
+
+	  # AMG smoother sweeps
+	  set amg smoother sweeps       = 2
+
+	  # amg smoother overlap
+	  set amg smoother overlap      = 1
+	end
+
+
+* The ``method`` parameter enables to choose an iterative solver for the linear system of equations. Lethe currently supports these core solution strategies :
+	* ``gmres`` (default parameter value), a GMRES iterative solver with ILU preconditioning.
+	* ``amg``, a GMRES iterative solver with AMG preconditioning and an ILU coarsener and smoother.
+	* ``bicgstab``, a BICGSTAB iterative solver with ILU preconditioning.
+	* ``direct``, a direct solver using `TrilinosWrappers::SolverDirect <https://www.dealii.org/current/doxygen/deal.II/classTrilinosWrappers_1_1SolverDirect.html>`_. 
+
+	.. tip:: 
+		On coarse meshes, the ``gmres``/``bicgstab`` solver with ILU preconditioning is more efficient. 
+
+		As the number of mesh elements increase, the ``amg`` solver is the most efficient. Generally, at 1M elements, the ``amg`` solver always outperforms the ``gmres`` or ``bicgstab`` for 2D steady-state problems.
+		
+		The use of ``direct`` solver should be avoided for 3D problems.
+
+* The ``verbosity`` option enables the display of the residual at each non-linear iteration, to monitor the progress of the linear iterations.
+
+.. note::
+	Example of a ``set verbosity = verbose`` output:
+
+	.. code-block:: text
+
+		-Tolerance of iterative solver is : 0.0429541
+		-Iterative solver took : 11 steps 
+		-Tolerance of iterative solver is : 3.62082e-05
+		-Iterative solver took : 16 steps 
+		-Tolerance of iterative solver is : 1.05775e-08
+		-Iterative solver took : 17 steps 
+		-Tolerance of iterative solver is : 1.00205e-12
+		-Iterative solver took : 16 steps 
+		-Tolerance of iterative solver is : 1e-13
+		-Iterative solver took : 5 steps 
+		-Tolerance of iterative solver is : 1e-13
+		-Iterative solver took : 0 steps 
+
+
+* The ``minimum residual`` parameter sets the value that the tolerance of the iterative solver should reach to proceed to the next iteration.
```

Comment:
This is not true. The residual to proceed onto the next iteration is the largest value between:
max (relative_residual * initial_residual , minimum_residual)
We set both a relative tolerance and an absolute tolerance. Otherwise we would solve the system very nicely for no reasons at all.

Reply:
The `max (relative_residual * initial_residual , minimum_residual)` sets the tolerance of iterative solver for this linear iteration (taking a given number of steps), but in the end we want the iterative solver to reach a tolerance that is below the `minimum_residual`, before proceeding to the next transient iteration, no?

Is this a matter of terminology between a linear solver iterative step (as in `Iterative solver took : 12 steps`), linear solver iteration (number of lines in the output), and simulation iteration(`Transient iteration` for a transient scheme)?

For instance, here I set my minimum residual to 1e-13, and we go to the next transient iteration only once the solver went all the way under this value:

<pre><code>
*****************************************************************
Transient iteration : 1        Time : 0.05     Time step : 0.05     CFL : 0       
*****************************************************************
  -Tolerance of iterative solver is : 0.0429541
  -Iterative solver took : 7 steps 
  -Tolerance of iterative solver is : 3.71444e-05
  -Iterative solver took : 10 steps 
  -Tolerance of iterative solver is : 1.14721e-08
  -Iterative solver took : 12 steps 
  -Tolerance of iterative solver is : 1.18663e-12
  -Iterative solver took : 12 steps 
  -Tolerance of iterative solver is : 1e-13
  -Iterative solver took : 3 steps 
  -Tolerance of iterative solver is : 1e-13
  -Iterative solver took : 0 steps 

*****************************************************************
Transient iteration : 2        Time : 0.1      Time step : 0.05     CFL : 0.715486
*****************************************************************
  -Tolerance of iterative solver is : 0.00406999
  -Iterative solver took : 8 steps 
  -Tolerance of iterative solver is : 3.64636e-06
  -Iterative solver took : 10 steps 
  -Tolerance of iterative solver is : 6.38355e-10
  -Iterative solver took : 11 steps 
  -Tolerance of iterative solver is : 1.15663e-13
  -Iterative solver took : 12 steps 
  -Tolerance of iterative solver is : 1e-13
  -Iterative solver took : 0 steps 
</code></pre>

Reply:
That is because your non-linear solver tolerance is low.
For example, if

> The `max (relative_residual * initial_residual , minimum_residual)` sets the tolerance of iterative solver for this linear iteration (taking a given number of steps), but in the end we want the iterative solver to reach a tolerance that is below the `minimum_residual`, before proceeding to the next transient iteration, no?

No this  is not correct. What controls if you proceed to the next iteration is the non-linear tolerance. If you set a non-linear solver tolerance of 1e-6 and a linear solver tolerance of 1e-13, with a relative tolerance of say 1e-3, you will never reach the tolerance of your iterative solver. The thing that decides if you move to the next iteration is always the non-linear solver tolerance. E.g. once your non-linear equation has been solved satisfactorily, you move on. The idea with the linear solver tolerance is that you don't want to solve your linear system of equations too acurately if you don't need to.

> 
> Is this a matter of terminology between a linear solver iterative step (as in `Iterative solver took : 12 steps`), linear solver iteration (number of lines in the output), and simulation iteration(`Transient iteration` for a transient scheme)?
> 
> For instance, here I set my minimum residual to 1e-13, and we go to the next transient iteration only once the solver went all the way under this value:
> 
> ```
> 
> *****************************************************************
> Transient iteration : 1        Time : 0.05     Time step : 0.05     CFL : 0       
> *****************************************************************
>   -Tolerance of iterative solver is : 0.0429541
>   -Iterative solver took : 7 steps 
>   -Tolerance of iterative solver is : 3.71444e-05
>   -Iterative solver took : 10 steps 
>   -Tolerance of iterative solver is : 1.14721e-08
>   -Iterative solver took : 12 steps 
>   -Tolerance of iterative solver is : 1.18663e-12
>   -Iterative solver took : 12 steps 
>   -Tolerance of iterative solver is : 1e-13
>   -Iterative solver took : 3 steps 
>   -Tolerance of iterative solver is : 1e-13
>   -Iterative solver took : 0 steps 
> 
> *****************************************************************
> Transient iteration : 2        Time : 0.1      Time step : 0.05     CFL : 0.715486
> *****************************************************************
>   -Tolerance of iterative solver is : 0.00406999
>   -Iterative solver took : 8 steps 
>   -Tolerance of iterative solver is : 3.64636e-06
>   -Iterative solver took : 10 steps 
>   -Tolerance of iterative solver is : 6.38355e-10
>   -Iterative solver took : 11 steps 
>   -Tolerance of iterative solver is : 1.15663e-13
>   -Iterative solver took : 12 steps 
>   -Tolerance of iterative solver is : 1e-13
>   -Iterative solver took : 0 steps 
> ```

Reply:
Ok, I'll remove that. If you want to add anything else, feel free to offer a formulation! Otherwise we can assume that it will be on the Theory documentation, for interested readers.